src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs

```diff
@@ -254,20 +254,10 @@ protected override SyntaxTrivia TogglePragmaDirective(SyntaxTrivia trivia)
             return SyntaxFactory.Trivia(newPragmaWarning);
         }
 
-        protected override SyntaxToken GetAdjustedTokenForPragmaDisable(SyntaxToken token, SyntaxNode root, TextLineCollection lines, int indexOfLine)
-        {
-            // normally we find the parent statement for the token but if the token is a close brace
-            // that could expand to a very large block, so we skip it and just return the token itself
-            if (token.IsKind(SyntaxKind.CloseBraceToken))
-            {
-                return token;
-            }
-
-            return base.GetAdjustedTokenForPragmaDisable(token, root, lines, indexOfLine);
-        }
-
         protected override SyntaxNode GetContainingStatement(SyntaxToken token)
-            => token.GetAncestor<StatementSyntax>();
+            // If we can't get a containing statement, such as for expression bodied members, then
+            // return the arrow clause instead
+            => (SyntaxNode)token.GetAncestor<StatementSyntax>() ?? token.GetAncestor<ArrowExpressionClauseSyntax>();
```

Comment:
A naming violation for a class declaration won't have StatementSyntax nor ArrowExpressionClauseSyntax.

```csharp
class camelCaseClassName{}
```